ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/restore/state/WordList.kt

```diff
@@ -1,21 +1,23 @@
 package co.electriccoin.zcash.ui.screen.restore.state
 
 import cash.z.ecc.sdk.model.SeedPhraseValidation
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.toPersistentList
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.map
 
 class WordList(initial: List<String> = emptyList()) {
-    private val mutableState = MutableStateFlow(initial)
+    private val mutableState: MutableStateFlow<ImmutableList<String>> = MutableStateFlow(initial.toPersistentList())
 
-    val current: StateFlow<List<String>> = mutableState
+    val current: StateFlow<ImmutableList<String>> = mutableState
 
     fun set(list: List<String>) {
-        mutableState.value = ArrayList(list)
+        mutableState.value = list.toPersistentList()
     }
 
     fun append(words: List<String>) {
-        mutableState.value = ArrayList(current.value) + words
+        mutableState.value = (ArrayList(current.value) + words).toPersistentList()
```

Comment:
It probably isn’t necessary here—I was thinking that + might mutate but I think it makes a copy.

Reply:
Yes, the PersistentList does a copy every time we want mutate it (add/remove item). So it's maybe not super efficient, but it's alright here, as we do a new list creation every time.

Reply:
You're right, it is less efficient.  I'm not worried too much about the performance here since users only interact with this once during onboarding, and the input is at human timescales.